fdbserver/storageserver.actor.cpp

```diff
@@ -4899,9 +4938,14 @@ ACTOR Future<Void> getKeyQ(StorageServer* data, GetKeyRequest req) {
 	// so we need to downgrade here
 	wait(data->getQueryDelay());
 
+	// Track time from requestTime through now as read queueing wait time
+	state double queueWaitEnd = g_network->timer();
+	data->counters.readQueueWaitSample.addMeasurement(queueWaitEnd - req.requestTime());
+
 	try {
 		Version commitVersion = getLatestCommitVersion(req.ssLatestCommitVersions, data->tag);
 		state Version version = wait(waitForVersion(data, commitVersion, req.version, req.spanContext));
+		data->counters.readVersionWaitSample.addMeasurement(g_network->timer() - queueWaitEnd);
```

Comment:
It looks like `getKey` is now the only request measured in the read latency sample that doesn't have its own operation specific sample. In practice this probably doesn't make a huge difference given that I think we only use get key requests to the SS if you bypass RYW, but at the same time that means having it would probably be low cost and would round out the coverage of the different ops contributing to the read latency sample.

I'll leave it up to you whether you want to include it.

Reply:
I had the same thought, in StorageMetrics I never see it actually used but that being the case it's not expensive to have other than logging an empty trace event often.  

What do you think of changing LatencySample to not log an event when the population size is 0?

Reply:
That's not a bad idea, though we'd have to handle that in status. I believe it is grabbing the most recent event for some of the latency samples and reporting the results from it, and we'd probably want it to detect this case where the event isn't current enough.

Reply:
Good point.  I was thinking a limit of 1 consecutive 0-population event could work, but then it might be harder to use for some telemetry destinations which extrapolate between data points.

I guess I'll just add the event and let it spam.